codeblock.py

```diff
@@ -0,0 +1,32 @@
+import re
+from typing import Text
+from block import Block
+
+
+class CodeBlock(Block):
+    '''
+        https://www.markdownguide.org/extended-syntax/#fenced-code-blocks
+        https://www.markdownguide.org/basic-syntax/#escaping-backticks
+
+        Use backticks(`) to indicate a code block.
+    '''
+    def _render(self):
+        return '<code>{}</code>'.format(self.content())
+
+    @staticmethod
+    def parse(content: Text):
+        '''
+        Parse a text content, and return (begin, end, Element) if parseable,
+        or (-1, -1, None) that no such a element
+        '''
+        code_pattern = r"(^`|[^\\]`)((([^\n])*)[^\\])*`"
```

Comment:
if you don't need group(3) and group(4), the regex can be simplified as 
```python
r"(^`|[^\\]`)([^\n]*[^\\])*`"
```

also, maybe the last asterisk is not necessary? What you want is make sure there is no back-slash before back-tick, then maybe this is good enough?
```python
r"(^`|[^\\]`)([^\n]*[^\\])`"
```

Reply:
That's necessary. without that, it can not parse ``, cause nothing inside code block.